src/databricks/labs/ucx/assessment/clusters.py

```diff
@@ -46,6 +47,8 @@ class ClusterInfo:
     creator: str | None = None
     """User-name of the creator of the cluster, if known."""
 
+    __id_attributes__: ClassVar[tuple[str, ...]] = ("cluster_id",)
```

Comment:
typically dunder signals Python builtin. I prefer only leading `__` underscore

Reply:
We can't use just a leading `__`: that makes it a private attribute, and it can't be accessed from outside the instance. (Dunder attributes are special: although they start with `__` that are not subject to mangling like ordinary fields. Arguably they signal a protocol in use behind the curtains, which indeed is what Python mainly uses them for.)

In this case alternatives are:

 - `_id_attributes: ClassVar[tuple[str, ...]]`: Protected, mildly cranky linter ("access to protected member of a client class").
 - `id_attributes: ClassVar[tuple[str, ...]]`: Public, which doesn't feel right for what is essentially metadata that we're trying to attach to the instance.

A bit further from home, is something like the SQLAlchemy approach where metadata (such as this) is placed in a sibling-class. That's a pretty large hammer to hit this particular nail though.

Any further thoughts?

Reply:
this confusion could have been avoided by using a method-based protocol...

Reply:
Let's stick with this for now